retriever.py

```diff
@@ -0,0 +1,272 @@
+                    latlon=True, markersize=4, zorder=100)
+            for i, (lon, lat, loc) in enumerate(zip(wp_lons, wp_lats, wp_locs)):
+                textlabel = "{:}   ".format(loc if loc else str(i))
+                x, y = bm(lon, lat)
+                plt.text(x, y, textlabel, **TEXT_CONFIG)
+
+            ax_bounds = plt.gca().bbox.bounds
+            width, height = int(round(ax_bounds[2])), int(round(ax_bounds[3]))
+            bbox = params['basemap']
+
+            if not init_time:
+                init_time = None
+
+            wms = wms_control.MSUIWebMapService(url,
+                                                username=config["WMS_login"][url][0],
+                                                password=config["WMS_login"][url][1],
+                                                version='1.1.1'
+                                                )
+            img = wms.getmap(layers=[layer],
+                             styles=[style],
+                             time=time,
+                             init_time=init_time,
+                             exceptions='application/vnd.ogc.se_xml',
+                             level=elevation,
+                             srs=config["predefined_map_sections"][section]["CRS"],
+                             bbox=(bbox['llcrnrlon'], bbox['llcrnrlat'], bbox['urcrnrlon'], bbox['urcrnrlat']),
+                             format='image/png',
+                             size=(width, height)
+                            )
+            image_io = io.BytesIO(img.read())
+            img = PIL.Image.open(image_io)
+
+            bm.imshow(img, interpolation="nearest", origin="upper")
+            bm.drawcoastlines()
+            bm.drawcountries()
+
+            fig.savefig(f"{flight}_{layer}.png")
+
+        # prepare vsec plots
+        path = [(wp[0], wp[1], datetime.datetime.now()) for wp in wps]
+        lats, lons = mslib.utils.coordinate.path_points(
+            [_x[0] for _x in path], [_x[1] for _x in path], numpoints=num_interpolation_points + 1, connection="greatcircle")
+        intermediate_indexes = []
+        ipoint = 0
+        for i, (lat, lon) in enumerate(zip(lats, lons)):
+            if abs(lat - wps[ipoint][0]) < 1E-10 and abs(lon - wps[ipoint][1]) < 1E-10:
+                intermediate_indexes.append(i)
+                ipoint += 1
+            if ipoint >= len(wps):
+                break
+
+        for url, layer, style in config["automated_plotting_vsecs"]:
+            fig.clear()
+
+            # setup ticks and labels
+            ax = fig.add_subplot(111, zorder=99)
+            ax.set_yscale("log")
+            p_bot, p_top = [float(x) * 100 for x in vertical.split(",")]
+            bbox = ",".join(str(x) for x in (num_interpolation_points, p_bot / 100, num_labels, p_top / 100))
+            ax.grid(b=True)
+            ax.patch.set_facecolor("None")
+            pres_maj = mslib.msui.mpl_qtwidget.MplSideViewCanvas._pres_maj
```

Comment:
a lot of the following is taken from the MplSideViewCanvas class (method redraw_yaxis) from
mslib/msui/mpl_qtwidget.py